### PR TITLE
Implement install requirements.txt cli

### PR DESCRIPTION
**Pull request by konstin**
Adds a very basic CLI to install from requirements.txt the same way that pip does. Works the same as pip for pydantic's `all.txt` for part i diffed. I've split the test refactoring necessary one stack up, not sure yet if that makes sense.

### CODE REVIEW

**Comment by konstin** [comment]
Current dependencies on/for this PR:
* main
  * **PR #43** <a href="https://app.graphite.dev/github/pr/konstin/poc-monotrail/43" target="_blank">Graphite</a> 
    * **PR #44** <a href="https://app.graphite.dev/github/pr/konstin/poc-monotrail/44" target="_blank">Graphite</a> 
      * **PR #45** <a href="https://app.graphite.dev/github/pr/konstin/poc-monotrail/45" target="_blank">Graphite</a>  👈
        * **PR #46** <a href="https://app.graphite.dev/github/pr/konstin/poc-monotrail/46" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/konstin/poc-monotrail/45?utm_source=stack-comment).

File: src/cli.rs [hunk 1]
```diff
@@ -246,19 +267,128 @@ fn poetry_install(
...
     Ok(())
 }
 
+///
+///
+/// The `venv` and `working_dir` options are to inject those for tests
+pub fn install(
+    requirements_files: Vec<String>,
+    _compile: bool,
+    _no_parallel: bool,
```
Comment (charliermarsh) [inline src/cli.rs:284]: Why unused? Should these just be removed until they're supported?

File: src/cli.rs [hunk 2]
```diff
@@ -246,19 +267,128 @@ fn poetry_install(
...
     )?;
     installed_done.append(&mut installed_new);
     Ok(())
 }
 
+///
+///
+/// The `venv` and `working_dir` options are to inject those for tests
+pub fn install(
+    requirements_files: Vec<String>,
```
Comment (charliermarsh) [inline src/cli.rs:282]: Does this need to take ownership?

**Review by charliermarsh — APPROVED** [review]
Looks reasonable. I'd tend towards not including unsupported options at all until they're supported, but it's a little bit of a stylistic preference.

Would you expect this installation to work for any requirements.txt file (assuming the deps are frozen, versioned, etc.)? What are the limitations? Does it work with source distributions? Git dependencies? Local dependencies?

**Pull request merged**